knimin/lib/tests/test_qiita_jira_util.py

```diff
@@ -1,24 +1,97 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The LabAdmin Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
 from unittest import TestCase, main
+from traceback import format_exc
+from functools import partial
 
-from knimin.lib.qiita_jira_util import create_project
+from knimin import qiita_client, jira_handler, db
+from knimin.lib.qiita_jira_util import create_study
 
 
 class TestQiitaJiraUtil(TestCase):
-    def test_create_project(self):
-        pj_name = 'My New Project'
-
-        # check success
-        pj, message = create_project(pj_name)
-        self.assertEqual(message, '')
-        self.assertEqual(pj['projectName'], pj_name)
-        self.assertEqual(pj['projectKey'], 'TM10001')
-
-        # check failure
-        pj, message = create_project(pj_name)
-        self.assertIsNone(pj)
-        exp_msg = ("A project with that name already exists., Project "
-                   "'My New Project' uses this project key.")
-        self.assertEqual(message, exp_msg)
+    def setUp(self):
+        self._clean_up_funcs = []
+
+    def tearDown(self):
+        for f in self._clean_up_funcs:
+            try:
+                f()
+            except Exception as e:
+                print("Database clean-up failed. Downstream tests might be "
+                      "affected by this! Reason: %s" % format_exc(e))
+
+    @classmethod
+    def tearDownClass(cls):
+        # Reset the qiita DB to make tests independent
+        qiita_client.post("/apitest/reset/")
```

Comment:
Did this get merged elsewhere? I'm not seeing it right now in the pr or in labadmin